social/app/models/node.py

```diff
@@ -105,8 +141,46 @@ def get_public_posts(self, page=1, size=50, next_url=None):
                 % url)
             return []
 
-    def create_or_update_remote_author(self, uuid):
-        response = self._get_author(uuid)
+    def create_or_update_remote_post(self, post_uuid):
+        response = self._get_post(post_uuid)
```

Comment:
`self._get_post()` doesn't do any key validation. Broken implementations will crash later on in this method when you attempt to access values from keys such as `posts`.